base/bool.jl

```diff
@@ -3,6 +3,8 @@
 # promote Bool to any other numeric type
 promote_rule(::Type{Bool}, ::Type{T}) where {T<:Number} = T
 
+convert(::Type{Bool}, x::Bool) = x   # prevent invalidation
```

Comment:
This one might not be necessary either, after the `Union{}` stuff is fixed. It's just a really common case that would otherwise fall back to `convert(::Type{T}, x::T)`.